### DISCUSSION TITLE
Navigation in tunnels

### DISCUSSION THREAD

**Discussion by umpfmpf** [devemux86/cruiser#535] [Ideas]
Would it be possible to continue simulating the last speed traveled in the tunnel?

**Comment by devemux86** [comment]
@umpfmpf thanks for the suggestion.

This could be possible, although it's more complicated than it seems.

**Comment by DIrgi** [comment]
Seems it was unfortunately of low interest so far.

Don't no much work it is - as you said - from my estimation
 it can work like: once GPS got lost and map says last spotting was in a tunnel or very close by, then go ahead with the last driven speed and if it was above speed limit, not faster then allowed speed as long as GPS is available again.

Would really help especially in high tunnel frequency areas and junctions right afterwards.
Cheers

**Reply by devemux86** [reply]
It's a bit of a complicated work. :slightly_smiling_face: 

If you disable rerouting, navigation continues immediately after the tunnel.

**Reply by DIrgi** [reply]
You as programmer probably now it the best, how work intensive it can be. Me for myself I see just the simplicity.
Since routing simulation is already working and existing it is then just something similar which has to be connected, continue navigating through tunnel where there's no GPS signal.
However maybe this can be a midterm project for the big version jump 6.0.

By the way, have you ever used meanwhile AI for doing the programming stuff?

**Reply by DIrgi** [reply]
I don't understand what you mean by:
"If you disable rerouting, navigation continues immediately after the tunnel."

Can't cruiser check GPS availability and then switch between modes?

Google maps can do it, Garmin as well (afaik)

**Reply by devemux86** [reply]
GPS detection is always available.

The question is why do you need navigation inside a tunnel, just enter / exit the tunnel and navigation continues.

**Reply by DIrgi** [reply]
Not always and sometimes not fast enough.
Especially when there are chains of tunnels, shortly interrupted but quit often still no signal available between these narrow valleys and/or not seldom nowadays that exitd are sometimes within tunnels (I.e. Switzerland) or right afterwards.
Let me say it in that way. When I use Google maps navigation for daily a to b navigation, I enjoy continuous routing through tunnels and especially what is coming soon on the screen and vice versa, I miss this feature in cruiser for smooth continuously navigation.

**Comment by umpfmpf** [comment]
I have to bring the issue out again. 

Cruiser has improved considerably in recent months, especially much faster. 

Now I'm back in the Alps with lots of tunnels. Unfortunately, I missed a descent directly to or between two tunnels a few times because Cruiser stopped with the loss of the GPS signal and I can no longer see the route. 

I would really like it if it were simply simulated at the speed allowed or the last speed driven.

**Reply by MotoUKRider** [reply]
Perhaps some kind of _Dead Reckoning_ mode, switched when the GPS is missed by `x` number of polls.  Like when you select a starting point for _Simulation_, this mode activates in the same way but from the last known GPS position and continues along the route until a GPS lock is obtained.

Describing the process is easy, of course!

**Reply by ramondevic** [reply]
Yes, I also think it’s very necessary to see the movement inside the tunnel to know where you are, especially if you have to exit right as the tunnel ends.

**Reply by Leeb64** [reply]
Es gibt öfters auch Abzweigungen in Tunneln. Da wärs schon hilfreich "in etwa" zu wissen wo man gerade sein könnte.

**Reply by Leeb64** [reply]
Was ist mit diesen "Bluetooth Tunnel Beacons"? Könnte man damit vielleicht was anfangen?

**Reply by DIrgi** [reply]
It should work somehow, because other Navi software can do it as well even when there's no GPS available or Software is connected to vehicle tachometer. I.e. Google Maps and Garmin can handle this.

**Reply by MotoUKRider** [reply]
I wonder if they use accelerometer sensor to "assume" the route is still being followed?
ie: you're still in motion regardless of GPS fix.

**Reply by DIrgi** [reply]
I thought this first as well Well but at the end acceleration is acceleration. There should be no acceleration when you ride continuously 80km/h. 
From my opinion it can be measured only then by GPS or mechanical via speedometer.
When you watch Google maps doing his job he's not as accurate as having GPS. Maybe it's a connection of last driven speed with acc influence.

AI says: Google maps does dead reckoning in Combi with map matching and BT beacons.

AI says: Garmin is using the last speed when you enter the tunnel to remain and continue.

So at least the last method could be a first trial für cruiser

**Reply by Leeb64** [reply]
Ob das mit dem Beschleunigungssensor funktioniet? Schwierig bei den Gerüttel.
Wie man den Tacho mit dem Smartphone verbinden will??
Das habe ich gefunden:
https://www.beaconzone.co.uk/blog/what-are-bluetooth-tunnel-beacons/

**Reply by Leeb64** [reply]
Ich denke auch das die "letzte" Methode .. + Korrektur duch diese Beacons(falls vorhanden) am sinnvollsten wäre.
